File: backend/src/Request/WarehouseFilterRequest.php
<?php

namespace App\Request;

class WarehouseFilterRequest
{
    private $typeOfCountry;

    private $cityName;

    private $countryID;

    /**
     * refers to the city name
     * Used this for get all warehouses in the country which the city exists in
     */
    private $target;

    private $countriesIDs;

    public function getTypeOfCountry()
    {
        return $this->typeOfCountry;
    }

    public function getCityName()
    {
        return $this->cityName;
    }

    public function getCountryID()
    {
        return $this->countryID;
    }

    public function getTarget()
    {
        return $this->target;
    }

    public function getCountriesIDs()
    {
        return $this->countriesIDs;
    }

    public function setCountriesIDs($countriesIDs)
    {
        $this->countriesIDs = $countriesIDs;
    }

}

File: flutter_boilerplate_dash/lib/module_warehouses/request/warehouse_filter_request.dart
class WarehouseFilterRequest {
  String? typeOfCountry;
  String? cityName;
  WarehouseFilterRequest(
      {
         this.typeOfCountry,
         this.cityName,
      });
  Map<String, dynamic> toJson() {
    return {
      'typeOfCountry': typeOfCountry,
      'target': cityName,
    };
  }
}